product/web3modal/src/main/kotlin/com/walletconnect/web3/modal/ui/components/ComponentDelegate.kt

```diff
@@ -0,0 +1,38 @@
+package com.walletconnect.web3.modal.ui.components
+
+import com.walletconnect.web3.modal.client.Web3Modal
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.onEach
+
+internal sealed class ComponentEvent {
+    object ModalHiddenEvent: ComponentEvent()
+    object ModalExpandedEvent: ComponentEvent()
+}
```

Comment:
In the Web3ModalState file, you check each of these values and return a boolean. 

You could also do
```suggestion
internal sealed class ComponentEvent(isOpen: Boolean) {
    object ModalHiddenEvent: ComponentEvent(false)
    object ModalExpandedEvent: ComponentEvent(true)
}
```

Then in Web3ModalSate where you listen to the event, you just grab the `isOpen` variable

Reply:
Why would we need it? I think just by an event itself it's clear on which state modal is

Reply:
Ah ok I see below